src/Services/Device.vala

```diff
@@ -227,7 +227,27 @@ public class Power.Services.Device : Object {
     }
 
     public string get_symbolic_icon_name_for_battery () {
-        return get_icon_name_for_battery () + "-symbolic";
+        if (!is_a_battery) {
+            return "preferences-system-power-symbolic";
+        }
+
+        var icon_name = "battery";
+
+        if (percentage >= 20) {
+            icon_name += "-%i".printf ((int) (5 * Math.round (percentage / 5)));
+        } else if (percentage >= 10) {
+            icon_name += "-10";
```

Comment:
Maybe this should be < 10. It feels way aggressive to have a single red line at 19% battery